compiler/noirc_evaluator/src/errors.rs

```diff
@@ -96,16 +99,21 @@ impl RuntimeError {
 impl From<RuntimeError> for FileDiagnostic {
     fn from(error: RuntimeError) -> FileDiagnostic {
         let call_stack = vecmap(error.call_stack(), |location| *location);
-        let diagnostic = error.into_diagnostic();
         let file_id = call_stack.last().map(|location| location.file).unwrap_or_default();
-
+        let diagnostic = error.into_diagnostic();
         diagnostic.in_file(file_id).with_call_stack(call_stack)
     }
 }
 
 impl RuntimeError {
     fn into_diagnostic(self) -> Diagnostic {
         match self {
+            RuntimeError::InternalError(InternalError::ReturnConstant { ref call_stack }) => {
+                let message = self.to_string();
+                let location =
+                call_stack.back().expect("Expected RuntimeError to have a location");
+               Diagnostic::simple_error(message, "constant value".to_string(), location.span)
```

Comment:
Actually, before we merge this, wouldn't this be more appropriate as a warning?

Reply:
In the linked issue it says that warning or error does not matter.
I went for errors because SSA pass does not support warning, but adding them should not be too difficult.
It is better I think to add warnings in a separate PR.

Reply:
@guipublic can you open an issue for this?